docs/development/new-cloud-provider-api-extension.md

```diff
@@ -0,0 +1,79 @@
+# API Extension and Validation for New Cloud Provider
+
+This document describes how to extend Gardener to support a new cloud provider in the API. There is a significant number of steps necessary.
+
+## Add Cloud Provider Support
+
+Cloud provider support is headed through the following major steps:
+
+1. Add the necessary structures, logic and tests to the code
+2. Update example templates
+3. Generate auto-generated code
+4. Generate examples
+
+The detailed steps are as follows:
+
+1. Add the cloud provider to the template files in [hack/templates/resources](../../hack/templates/resources/) :
+  * `CloudProfile` [30-cloudprofile.yaml.tpl](../../hack/templates/resources/30-cloudprofile.yaml.tpl) This is the definition of a profile for this provider.
+  * `Secret` [40-secret-seed.yaml.tpl](../../hack/templates/resources/40-secret-seed.yaml.tpl) This contains the secrets necessary to invoke the cloud provider's API to manage Shoot ectd backups, and the kubeconfig to access the Seed cluster. 
+  * `Seed`: [50-seed.yaml.tpl](../../hack/templates/resources/50-seed.yaml.tpl) This contains the configuration for the Seed cluster.. 
+  * `Secret`: [70-secret-cloudprovider.yaml.tpl](../../hack/templates/70-secret-cloudprovider.yaml.tpl) This contains the secrets necessary for to invoke the cloud provider's API to deploy the Shoot cluster(s).
+  * `SecretBinding`: [80-secretbinding-cloudprovider.yaml.tpl](../../hack/templates/resources/80-secretbinding-cloudprovider.yaml.tpl) This binds the secret defined in the previous step with quotas from the current or another namespace.
```

Comment:
I would leave the quota part out to reduce surface for confusion - we will provide separate documentation on how exactly the `SecretBinding` resource is used. Let's focus here on the cloud provider extension part.